### PR TITLE
IDLE: Simplify DynOptionsMenu __init__code

### PR DESCRIPTION
**Pull request by JosephSBoyle**
In dynoption.py, don't copy the kwargs dict and do use subscript syntax for config.

## Testing
To test the changes don't break anything, simply set the 'highlightthickness' value:

```python
    dyn = DynOptionMenu(top,var, "old1","old2","old3","old4", highlightthickness=5)
   #                                                          ^^^^^^^^^^^^^^^^^^^^      
```
and run the script:

```bash
$ python Lib/idlelib/dynoption.py
```

### CODE REVIEW

File: Lib/idlelib/dynoption.py [hunk 1]
```diff
@@ -2,24 +2,18 @@
...
-        kwargsCopy=copy.copy(kwargs)
-        if 'highlightthickness' in list(kwargs.keys()):
-            del(kwargs['highlightthickness'])
+        highlightthickness = kwargs.pop('highlightthickness', None)
         OptionMenu.__init__(self, master, variable, value, *values, **kwargs)
-        self.config(highlightthickness=kwargsCopy.get('highlightthickness'))
-        #self.menu=self['menu']
-        self.variable=variable
-        self.command=kwargs.get('command')
+        self.config(highlightthickness=highlightthickness)
```
Comment (terryjreedy) [inline Lib/idlelib/dynoption.py]: For IDLE, I use  dict syntax for single-attribute configure.
```suggestion
        self['highlightthickness'] = highlightthickness
```

**Review by terryjreedy — COMMENTED** [review]
#71942 proposes to eliminate this file, but I plan to apply this now with the change in the comment and replacement of the htest line with
`    dyn = DynOptionMenu(top,var, "old1","old2","old3", highlightthickness=0)`

The efficiency gain I care about is in reading the code ;-)

**Comment by JosephSBoyle** [comment]
Hi @terryjreedy, I've applied the patch you suggested.
Let me know if there's anything else I can do!:)

**Review by terryjreedy — APPROVED** [review]
I improved init docstring and added highlightthickness to htest and note as to why no unittests.

**Comment by terryjreedy** [comment]
Backports merged.

**Pull request merged**